### PR TITLE
[ADD] describing fields in `event.json` file

### LINKED ISSUE #1254: Add descriptions for onset and duration in `event.json` file
### Describe the problem

As currently implemented, the `events.json` file being written does not contain information about the two required columns "onset" and "duration" in the `.json` file.

As those columns are REQURIED, I would propose adding some background information to the `.json` file every time events are written for people who are not familiar with the structure of the `events.tsv` files in.REQUIRED

### Describe your solution

Addiing this to the _events_json_ function

```
  "onset": {
        "Requirement Level": "REQUIRED",
        "Data type": "number",
        "Description": (
            "Onset (in seconds) of the event from the beginning of the first data point. "
            "Negative onsets account for events before the first stored data point. "
            "This column must appear first in the file."
        )
    },
    "duration": {
        "Data type": "number or 'n/a'",
        "Description": (
            "Duration of the event in seconds from onset. Must be zero, positive, or 'n/a' if unavailable. "
            "A zero value indicates an impulse event.\n"
            "This column must appear second in the file."
        )
    }

```

Adding

### Describe possible alternatives

As the documentation about this is specific on the BIDS website, it does not has to be implemented.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by JuliusWelzel**
PR Description
--------------
As currently implemented, the events.json file being written does not contain information about the two required columns "onset" and "duration" in the .json file.

As those columns are REQURIED, I would propose adding some background information to the .json file every time events are written for people who are not familiar with the structure of the events.tsv files in.REQUIRED.

See https://github.com/mne-tools/mne-bids/issues/1254 for details.

- closes #1254 

Merge checklist
---------------

Maintainer, please confirm the following before merging.
If applicable:

- [ ] All comments are resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] New contributors have been added to [CITATION.cff](https://github.com/mne-tools/mne-bids/blob/main/CITATION.cff)
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: mne_bids/write.py [hunk 1]
```diff
@@ -338,7 +338,28 @@ def _events_json(fname, overwrite=False):
...
     """
     new_data = {
-        "sample": {"Description": "The event onset time in number of sampling points."},
+        "onset": {
+            "Requirement Level": "REQUIRED",
+            "Data type": "number",
+            "Description": (
+                "Onset (in seconds) of the event from the beginning of the first data point. "
+                "Negative onsets account for events before the first stored data point. "
+                "This column must appear first in the file."
```
Comment (hoechenberger) [inline mne_bids/write.py:347]: I don't think we need this here, as we're handling the column order ourselves

File: mne_bids/write.py [hunk 2]
```diff
@@ -338,7 +338,28 @@ def _events_json(fname, overwrite=False):
...
+            ),
+        },
+        "duration": {
+            "Requirement Level": "REQUIRED",
+            "Data type": "number or 'n/a'",
+            "Description": (
+                "Duration of the event in seconds from onset. Must be zero, positive, or 'n/a' if unavailable. "
+                "A zero value indicates an impulse event.\n"
+                "This column must appear second in the file."
```
Comment (hoechenberger) [inline mne_bids/write.py:356]: I don't think we need this here, as we're handling the column order ourselves

File: mne_bids/write.py (hunk 1 above)
Comment (JuliusWelzel) [reply mne_bids/write.py]: Handled in https://github.com/mne-tools/mne-bids/pull/1255/commits/8895caff522d055ae25d2fcf55b833eccb2eda85

File: mne_bids/write.py (hunk 2 above)
Comment (JuliusWelzel) [reply mne_bids/write.py]: Handled in https://github.com/mne-tools/mne-bids/pull/1255/commits/8895caff522d055ae25d2fcf55b833eccb2eda85

File: mne_bids/write.py [hunk 3]
```diff
@@ -338,7 +338,28 @@ def _events_json(fname, overwrite=False):
         Whether to overwrite the output file if it exists.
     """
     new_data = {
-        "sample": {"Description": "The event onset time in number of sampling points."},
+        "onset": {
+            "Requirement Level": "REQUIRED",
+            "Data type": "number",
```
Comment (sappelhoff) [inline mne_bids/write.py:343]: Neither "Requirement Level" nor "Data type" are needed here, I assume that you got them from the BIDS schema, but please remove them from this PR :-)

IMHO we only need "Description" and "Units" (the latter of which is currently not added in this PR)

Example can be found here: https://github.com/bids-standard/bids-examples/blob/6470461ce3c357080e39a11d7d7b1c4cde9f567b/eeg_matchingpennies/task-matchingpennies_events.json#L2-L8

File: mne_bids/write.py [hunk 4]
```diff
@@ -338,7 +338,28 @@ def _events_json(fname, overwrite=False):
...
+            "Data type": "number or 'n/a'",
+            "Description": (
+                "Duration of the event in seconds from onset. "
+                "Must be zero, positive, or 'n/a' if unavailable. "
+                "A zero value indicates an impulse event. "
+            ),
+        },
+        "sample": {
+            "Data type": "number",
+            "Description": ("The event onset time in number of sampling points."),
```
Comment (sappelhoff) [inline mne_bids/write.py:361]: Perhaps you could double-check this, but I think our "sample" column is zero indexed.

Perhaps you could say so in the Description here.

- xref https://github.com/bids-standard/bids-specification/pull/1457

**Review by sappelhoff — COMMENTED** [review]
Thanks @JuliusWelzel I merged main into your branch, so please pull before continuing to work on this (main contains some CI fixes now).

As for your PR, I am happy with the proposed changes, but please observe my comments in the code.

Finally, please follow the steps in the second to last section of this document: https://github.com/mne-tools/mne-bids/blob/main/CONTRIBUTING.md#instructions-for-first-time-contributors

File: mne_bids/write.py (hunk 4 above)
Comment (JuliusWelzel) [reply mne_bids/write.py]: Not sure if I understand the discussion correctly, but as I read it, the user can decide how a sample is used. I now added that the first value is supposed to be 0.

File: CITATION.cff [hunk 5]
```diff
@@ -264,6 +264,10 @@ preferred-citation:
       family-names: Rockhill
       affiliation: 'University of Oregon, Eugene OR, USA'
       orcid: 'https://orcid.org/0000-0003-3868-7453'
+    - given-names: Julius
+      family-names: Welzel
+      affiliation: 'Department of Neurology, Kiel University, Germany'
+      orcid: 'https://orcid.org/0000-0003-4782-5360'
```
Comment (sappelhoff) [inline CITATION.cff:270]: this should be in the "authors" section above, not in the `preferred_citation.authors` section here :-) line 178 in this current file looks like the right place.

**Review by sappelhoff — APPROVED** [review]
one last thing to do, the rest looks great -- thanks

File: CITATION.cff (hunk 5 above)
Comment (JuliusWelzel) [reply CITATION.cff]: Ups, solved in https://github.com/mne-tools/mne-bids/pull/1255/commits/3a37466eaea68f7d179c600ac75aa4a7c4f7b1b8

**Comment by sappelhoff** [comment]
Thanks @JuliusWelzel!

**Pull request merged**